src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCommon.java

```diff
@@ -0,0 +1,299 @@
+    }
+
+    /**
+     * Adds metadata for the given column in the batch/file.
+     * 
+     * @param positionInSource
+     *            Indicates which column the metadata is for. Columns start at 1.
+     * @param name
+     *            Name for the column (optional if only using column ordinal in a mapping for SQLServerBulkCopy operation)
+     * @param jdbcType
+     *            JDBC data type of the column
+     * @param precision
+     *            Precision for the column (ignored for the appropriate data types)
+     * @param scale
+     *            Scale for the column (ignored for the appropriate data types)
+     * @param dateTimeFormatter
+     *            format to parse data that is sent
+     * @throws SQLServerException
+     *             when an error occurs
+     */
+    void addColumnMetadataInternal(int positionInSource,
+            String name,
+            int jdbcType,
+            int precision,
+            int scale,
+            DateTimeFormatter dateTimeFormatter) throws SQLServerException {
+        loggerExternal.entering(loggerClassName, "addColumnMetadata", new Object[] {positionInSource, name, jdbcType, precision, scale});
+
+        String colName = "";
+
+        if (0 >= positionInSource) {
+            MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_invalidColumnOrdinal"));
+            Object[] msgArgs = {positionInSource};
+            throw new SQLServerException(form.format(msgArgs), SQLState.COL_NOT_FOUND, DriverError.NOT_SET, null);
+        }
+
+        if (null != name)
+            colName = name.trim();
+        else if ((columnNames != null) && (columnNames.length >= positionInSource))
+            colName = columnNames[positionInSource - 1];
+
+        if ((columnNames != null) && (positionInSource > columnNames.length)) {
+            MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_invalidColumn"));
+            Object[] msgArgs = {positionInSource};
+            throw new SQLServerException(form.format(msgArgs), SQLState.COL_NOT_FOUND, DriverError.NOT_SET, null);
+        }
+
+        checkDuplicateColumnName(positionInSource, name);
+        switch (jdbcType) {
+            /*
+             * SQL Server supports numerous string literal formats for temporal types, hence sending them as varchar with approximate
+             * precision(length) needed to send supported string literals. string literal formats supported by temporal types are available in MSDN
+             * page on data types.
+             */
+            case java.sql.Types.DATE:
+            case java.sql.Types.TIME:
+            case java.sql.Types.TIMESTAMP:
+            case microsoft.sql.Types.DATETIMEOFFSET:
+                if (this instanceof SQLServerBulkCSVFileRecord) {
+                    columnMetadata.put(positionInSource, new ColumnMetadata(colName, jdbcType, 50, scale, dateTimeFormatter));
```

Comment:
Both `if` and `else` blocks call `columnMetadata.put(positionInSource, new ColumnMetadata(colName, jdbcType, 50, scale, dateTimeFormatter));`. Also why do we have 50 as precision?

Reply:
Sure. I agree that the if and else statement having the same is redundant. The 50 precision is used for CSV here since we can't rely on the precision from the SQL server - temporal types can be stored in CSV using any of the supported string literal format, and we pass it as varchar/nvarchar to server.